rfcs/0000-kuadrant_sub_components_configurations.md

```diff
@@ -0,0 +1,237 @@
+kind: Kuadrant
+metadata:
+    name: kuadrant-sample
+spec:
+    limitador:
+        afffinity: ...
+        listener: ...
+        pdb: ...
+        replicas: ...
+        resourceRequirements: ...
+        storage: ...
+    authorino:
+        eveluatorCacheSize: ...
+        healthz: ...
+        listener: ...
+        logLevel: ...
+        metrics: ...
+        oidcServer: ...
+        replicas: ...
+        tracing: ...
+        volumes: ...
+status:
+    ...
+```
+
+The Kuadrant operator will watch for changes in the Authorino and Limitador CRs, reconciling back any changes that a user may do to these configurations.
+How ever Kuadrant operator will not reconcile fields that are given above. 
+An example of this is the `image` field on the Authorino CR.
+This field allows a user to set the image that Authorino is deployed with.
+The feature is meant for dev and testing purposes. 
+If a user wishes to use a different image, they can.
+Kuadrant assumes they know what they are doing but requires the user to set the change on the component directly.
+
+Only the sub component operator will be responsible for actioning the configurations pasted form the Kuadrant CR to the sub components CR.
+This ensure no extra changes will be required in the sub operators to meet the needs of Kuadrant.
+
+Status errors related to the configuration of the sub components should be reported back to the Kuadrant CR.
+The errors messages in Kuadrant state what components are currently having issue and which resource to review for more details.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+All the fields in the Authorino and Limitador CRs that are configurable in the Kuadrant CR are optional and have sound defaults.
+Kuadrant needs to remain installable with out having to set any spec in the Kuadrant CR.
+
+The Kuadrant operator should only reconcile the spec that is given. 
+This would mean if the user states the number of replicas to be used in one of the components only the replica field for that component should be reconciled.
+As the other fields would be blank at this stage, blank fields would not be reconciled to the component CR.
+By this behaviour a few things are being achieved.
+Component controllers define the defaults to be used in the components.
+Optional fields in the component CRs never get set with blank values.
+Blank values in the component CR could override the defaults of the components causing unexpected behaviour.
+Existing Kuadrant users may already have custom fields set in the component CRs.
+By only reconciling the fields set in the kuadrant CR this allows time for a user to migrate their custom configuration from the component CR to the Kuadrant CR.
+
+## Fields to reconcile
+Fields being reconcile can be classified into different groups.
+These classifications are based around the tasks a user is achieve.
+
+- Kubernetes native, setting that affect how Kubernetes handles the resource.
```

Comment:
I wonder about labels. Are those in this group? Or maybe they could be propagated from the labels of the Kuadrant CR itself. If this is the case, are labels out of scope of the RFC?

The ability to set labels in the components' CRs can be useful for selecting not only these custom resources in particular, but also other resources that the corresponding component operator creates based on those. This is the case of Authorino, whose operator creates services propagating the same labels applied to the Authorino CR, including, e.g., Authorino's metrics service, later used by the scraper to find the service.

Reply:
Labels are interesting, and would be out of scope of the RFC. This RFC is focusing on  the spec of CRs. 

If the labels were to be managed you would also need to take into account other updater's of the labels on the resources. The source of truth for the labels becomes a bit muddy.

Reply:
One thing is for sure. We need a way for users to be able to propagate labels across resources initially created by our controllers.

That said, I can live with labels being out of the scope of this RFC. It should be clear then for one who implements this RFC that reconciling the resources back to the desired state declared in the `Kuadrant` CR should NOT include the labels (at least not because of this RFC.)